safe-global/safe-wallet-web
# [Counterfactual] Deploy safe with first transaction

## What it solves

Resolves #3184 

## How this PR fixes it

- Deploys safe with first transaction
- Shows gas estimation for the bundle
- Clear counterfactual state after deployment
- Hide nonce in all transaction flows

## ToDos

- [x] Adjust AppFrame to handle incoming requests for counterfactual safes

## How to test it

1. Create a counterfactual 1/1 safe
2. Send some funds to it
3. Create a transaction
4. Gas estimation should be displayed
5. Execute the transaction
6. Observe you have to sign first
7. Observe the safe is deployed after execution
8. Observe the tx history is visible after execution

## Screenshots

## Checklist
